lib/decode.js

```diff
@@ -166,6 +184,42 @@ function _getSuffix({cborldBytes}) {
   const tagValue = cborldBytes[index];
   let registryEntryId;
   if(isModern) {
+    // structure is fixed with new tag processing rules
```

Comment:
Does this need a check for the 2 element array 0x82 at index 3?

Tricky CBOR parsing of untrusted inputs here seems unfortunate.  In this "modern" mode it's looking for array of two elements right?  And the first is the registry id.  Perhaps this could be a simpler check for only 0x82 and the id value when uncompressed.  That's only a byte I think?  Then refactor so the calling code would know it's in this mode and use the real CBOR decoder for the whole array using the compression flag as needed.